core/src/main/java/io/questdb/griffin/engine/functions/bind/BindVariableServiceImpl.java

```diff
@@ -245,7 +245,7 @@ public void setBin(int index) throws SqlException {
     public void setBin(int index, BinarySequence value) throws SqlException {
         indexedVariables.extendPos(index + 1);
         Function function = indexedVariables.getQuick(index);
-        if (function == null) {
+        if (function == null || function == UndefinedFunction.INSTANCE) {
```

Comment:
In case of other column types we fall into the `indexedVariables.setQuick()` branch when the `function` is `UndefinedFunction.INSTANCE` while for binary type we throw a `SqlException`. Is that on purpose?

Reply:
the logic is the same for `setBin()` as for other types. It's expressed slightly differently for some reason, but it's the same logic. 

the original intent was this: if a function at this index was not defined at all then create a new one. otherwise, try the use the existing one. 

the new intent is this: if a function at this index was not defined at all **OR it was defined with an unknown type** then create a new one.  otherwise, try the use the existing one.

Reply:
My bad, I misread the branches.